src/calculators/BinaryVotingPowerEarningPowerCalculator.sol

```diff
@@ -0,0 +1,210 @@
+                        Immutable Storage
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice The voting power token address.
+  address public immutable VOTING_POWER_TOKEN;
+
+  /// @notice The block number at which the voting power update starts.
+  uint48 public immutable SNAPSHOT_START_BLOCK;
+
+  /*///////////////////////////////////////////////////////////////
+                          Storage
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice The time interval after which the voting power is updated.
+  uint48 public votingPowerUpdateInterval;
+
+  /// @notice The oracle eligibility module that determines if a delegate qualifies for earning
+  /// power.
+  IOracleEligibilityModule public oracleEligibilityModule;
+
+  /*///////////////////////////////////////////////////////////////
+                            Constructor
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Initializes the BinaryVotingPowerEarningPowerCalculator.
+  /// @param _owner The owner of the contract.
+  /// @param _oracleEligibilityModule The oracle eligibility module address.
+  /// @param _votingPowerToken The voting power token address.
+  /// @param _votingPowerUpdateInterval The voting power update interval.
+  constructor(
+    address _owner,
+    address _oracleEligibilityModule,
+    address _votingPowerToken,
+    uint48 _votingPowerUpdateInterval
+  ) Ownable(_owner) {
+    if (_votingPowerToken == address(0)) {
+      revert BinaryVotingPowerEarningPowerCalculator__InvalidAddress();
+    }
+
+    VOTING_POWER_TOKEN = _votingPowerToken;
+    SNAPSHOT_START_BLOCK = uint48(block.number);
+
+    _setOracleEligibilityModule(_oracleEligibilityModule);
+    _setVotingPowerUpdateInterval(_votingPowerUpdateInterval);
+  }
+
+  /*///////////////////////////////////////////////////////////////
+              Proxy Eligibility Module Functions
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Gets the earning power of a voter.
+  /// @param _staker The address whose voting power is being calculated.
+  /// @return _votingPower The voting power of the voter.
+  /// @dev This function uses the voter's snapshot voting power at the most recent
+  ///      voting power update interval, and returns zero if the delegatee is
+  ///      ineligible **unless** the oracle is paused or stale — in which case,
+  ///      it still returns the snapshot voting power.
+  function getEarningPower(uint256, /* _amountStaked */ address _staker, address /* _delegatee */ )
+    external
+    view
+    returns (uint256 _votingPower)
+  {
+    _votingPower = _getEarningPower(_staker);
+  }
+
+  /// @notice Gets the new earning power of a voter.
+  /// @param _staker The address whose voting power is being calculated.
+  /// @return _votingPower The voting power of the voter.
+  /// @return _isQualifiedForBump Whether the voter is qualified for a bump.
```

Comment:
We should explain that _staker will always be qualfied for an earning power bump. We rely on the update interval to prevent frequent bumping.